lib/roo/excelx/cell/number.rb

```diff
@@ -15,14 +15,13 @@ def initialize(value, formula, excelx_type, style, link, coordinate)
 
         def create_numeric(number)
           return number if Excelx::ERROR_VALUES.include?(number)
-
           case @format
           when /%/
             Float(number)
           when /\.0/
             Float(number)
           else
-            number.include?('.') ? Float(number) : Integer(number)
+            (number.include?(".") || (/\A\d+E[-+]\d+\z/ =~ number)) ? Float(number) : Integer(number)
```

Comment:
Hi, I locally patched this regular expression with a `-?` in front of the first `\d+`, so it also supports negative numbers in scientific notation (which I encountered, upon which I found the issue this pull request solves).

I can imagine handling a pull request is more work than just changing it yourself, but that depends on your workflow. Shall I make a pull request for it?